Refactoring doesn't work in 2024.2

### Description of the bug:


In the latest 2024.2 RC, when running a refactor operation that triggers IntelliJ to "find usages", such as when renaming a symbol, it is failing due to a `NoClassDefFoundError` coming from the Bazel Intellij plugin.

### Which category does this issue belong to?

Intellij

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.


Open a Java project, try to rename a symbol using IntelliJ refactoring tools, observe bug.

### Which Intellij IDE are you using? Please provide the specific version.


2024.2 RC

### What programming languages and tools are you using? Please provide specific versions.


Java/Kotlin

### What Bazel plugin version are you using?


2024.07.16.0.1-api-version-242

### Have you found anything relevant by searching the web?

I think we have to explicitly declare a dependency in plugin.xml on `com.intellij.modules.platform`, but I'm not fully sure. This class still exists in IntelliJ: https://github.com/JetBrains/intellij-community/blob/master/platform/lvcs-impl/src/com/intellij/history/core/Paths.java

### Any other information, logs, or outputs that you want to share?

```
2024-08-01 16:36:40,960 [  44070] SEVERE - #c.i.r.BaseRefactoringProcessor - com/intellij/history/core/Paths
java.lang.NoClassDefFoundError: com/intellij/history/core/Paths
	at com.google.idea.blaze.base.lang.buildfile.search.BlazePackage.getPackageRelativePath(BlazePackage.java:117)
	at com.google.idea.blaze.base.lang.buildfile.references.LabelUtils.createLabelForFile(LabelUtils.java:41)
	at com.google.idea.blaze.base.lang.buildfile.search.BuildReferenceSearcher.processFileReferences(BuildReferenceSearcher.java:106)
	at com.google.idea.blaze.base.lang.buildfile.search.BuildReferenceSearcher.processQuery(BuildReferenceSearcher.java:61)
	at com.google.idea.blaze.base.lang.buildfile.search.BuildReferenceSearcher.processQuery(BuildReferenceSearcher.java:42)
	at com.intellij.openapi.application.QueryExecutorBase.execute(QueryExecutorBase.java:76)
	at com.intellij.util.ExecutorsQuery.processResults(ExecutorsQuery.java:30)
	at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:83)
	at com.intellij.util.AbstractQuery.delegateProcessResults(AbstractQuery.java:100)
	at com.intellij.util.MergeQuery.processResults(MergeQuery.java:22)
	at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:83)
	at com.intellij.util.AbstractQuery.delegateProcessResults(AbstractQuery.java:100)
	at com.intellij.util.UniqueResultsQuery.processResults(UniqueResultsQuery.java:37)
	at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:83)
	at com.intellij.util.AbstractQuery.forEach(AbstractQuery.java:75)
	at com.intellij.util.AbstractQuery.findAll(AbstractQuery.java:24)
	at com.intellij.refactoring.rename.RenamePsiElementProcessorBase.findReferences(RenamePsiElementProcessorBase.java:82)
	at com.intellij.refactoring.rename.RenamePsiFileProcessor.findReferences(RenamePsiFileProcessor.java:44)
	at com.intellij.refactoring.rename.RenameUtil.processUsages(RenameUtil.java:113)
	at com.intellij.refactoring.rename.RenameUtil.findUsages(RenameUtil.java:76)
	at com.intellij.refactoring.rename.RenameProcessor.findUsages(RenameProcessor.java:295)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runReadAction(AnyThreadWriteThreadingSupport.kt:264)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runReadAction(AnyThreadWriteThreadingSupport.kt:221)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:857)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:66)
	at com.intellij.refactoring.BaseRefactoringProcessor.lambda$doRun$0(BaseRefactoringProcessor.java:187)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:328)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:477)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:133)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressSynchronously$9(CoreProgressManager.java:567)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$new$0(ProgressRunner.java:85)
	at com.intellij.concurrency.client.ClientIdPropagation.captureClientIdInFunction$lambda$9(ClientIdPropagation.kt:83)
	at com.intellij.concurrency.client.ClientIdPropagation.captureClientIdInFunction$lambda$9(ClientIdPropagation.kt:83)
	at com.intellij.concurrency.client.ClientIdPropagation.captureClientIdInFunction$lambda$9(ClientIdPropagation.kt:83)
	at com.intellij.concurrency.client.ClientIdPropagation.captureClientIdInFunction$lambda$9(ClientIdPropagation.kt:83)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:249)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:96)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:221)
	at com.intellij.platform.diagnostic.telemetry.helpers.TraceKt.use(trace.kt:49)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:220)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:660)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:735)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:691)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:659)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:79)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:202)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:96)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:249)
	at com.intellij.openapi.progress.impl.ProgressRunner$ProgressRunnable.run(ProgressRunner.java:501)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$18(ProgressRunner.java:466)
	at com.intellij.util.concurrency.ChildContext$runAsCoroutine$1.invoke(propagation.kt:92)
	at com.intellij.util.concurrency.ChildContext$runAsCoroutine$1.invoke(propagation.kt:92)
	at com.intellij.util.concurrency.ChildContext.runAsCoroutine(propagation.kt:97)
	at com.intellij.util.concurrency.ChildContext.runAsCoroutine(propagation.kt:92)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$19(ProgressRunner.java:464)
	at com.intellij.concurrency.client.ClientIdPropagation.captureClientIdInRunnable$lambda$3(ClientIdPropagation.kt:43)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:735)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:732)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:732)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.ClassNotFoundException: com.intellij.history.core.Paths PluginClassLoader(plugin=PluginDescriptor(name=Bazel for IntelliJ, id=com.google.idea.bazel.ijwb, descriptorPath=plugin.xml, path=~/.config/JetBrains/RemoteDev-IU-242.20224.159/_home_adam_miskiewicz_repos_treehouse/plugins/ijwb, version=2024.07.16.0.1-api-version-242, package=null, isBundled=false), packagePrefix=null, state=active, parents=PluginDescriptor(name=Terminal, id=org.jetbrains.plugins.terminal, moduleName=intellij.terminal.sh, descriptorPath=intellij.terminal.sh.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/terminal, version=242.20224.159, package=org.jetbrains.plugins.terminal.sh, isBundled=true), PluginDescriptor(name=Terminal, id=org.jetbrains.plugins.terminal, moduleName=intellij.terminal/cloud, descriptorPath=intellij.terminal.cloud.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/terminal, version=242.20224.159, package=org.jetbrains.plugins.terminal.cloud, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.features-trainer, descriptorPath=kotlin.features-trainer.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.training, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.project-wizard.maven, descriptorPath=kotlin.project-wizard.maven.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.tools.projectWizard.maven, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.project-wizard.idea.k1, descriptorPath=kotlin.project-wizard.idea.k1.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.tools.projectWizard.k1, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.project-wizard.gradle, descriptorPath=kotlin.project-wizard.gradle.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.tools.projectWizard.gradle, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.project-wizard.idea, descriptorPath=kotlin.project-wizard.idea.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.tools.projectWizard, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.project-wizard.compose, descriptorPath=kotlin.project-wizard.compose.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.tools.composeProjectWizard, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.onboarding.gradle, descriptorPath=kotlin.onboarding.gradle.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.onboarding.gradle, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.onboarding, descriptorPath=kotlin.onboarding.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.onboarding, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin-ultimate.javascript.nodeJs, descriptorPath=kotlin-ultimate.javascript.nodeJs.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.ultimate.nodejs, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin-ultimate.javascript.debugger, descriptorPath=kotlin-ultimate.javascript.debugger.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.ultimate.javascript.debugger, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.run-configurations.testng, descriptorPath=kotlin.run-configurations.testng.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.testng, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.code-insight.structural-search.k1, descriptorPath=kotlin.code-insight.structural-search.k1.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.structuralsearch, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-reference-index, descriptorPath=kotlin.compiler-reference-index.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.search.refIndex, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.run-configurations.jvm, descriptorPath=kotlin.run-configurations.jvm.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.runConfigurations.jvm, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin-ultimate.profiler, descriptorPath=kotlin-ultimate.profiler.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.profiler, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.ml-completion, descriptorPath=kotlin.ml-completion.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.mlCompletion, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.compiler-plugin-support.maven, descriptorPath=kotlin.compiler-plugins.compiler-plugin-support.maven.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.maven.compilerPlugin, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.maven, descriptorPath=kotlin.maven.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.maven, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.code-insight.live-templates.k1, descriptorPath=kotlin.code-insight.live-templates.k1.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.liveTemplates.k1, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.code-insight.live-templates.shared, descriptorPath=kotlin.code-insight.live-templates.shared.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.liveTemplates, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.code-insight.inspections.k1, descriptorPath=kotlin.code-insight.inspections.k1.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.k1.codeinsight.inspections, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.code-insight.k1, descriptorPath=kotlin.code-insight.k1.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.k1.codeinsight, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.jvm-decompiler, descriptorPath=kotlin.jvm-decompiler.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.jvmDecompiler, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.run-configurations.junit, descriptorPath=kotlin.run-configurations.junit.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.junit, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.j2k.k1.new.post-processing, descriptorPath=kotlin.j2k.k1.new.post-processing.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.j2k.post.processing, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.j2k.k1.old.post-processing, descriptorPath=kotlin.j2k.k1.old.post-processing.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.j2k.old.post.processing, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.injection.k1, descriptorPath=kotlin.injection.k1.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.injection, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.i18n, descriptorPath=kotlin.i18n.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.i18n, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.gradle.code-insight-groovy, descriptorPath=kotlin.gradle.code-insight-groovy.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.groovy, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.grazie, descriptorPath=kotlin.grazie.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.grazie, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.gradle.gradle-tooling, descriptorPath=kotlin.gradle.gradle-tooling.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.gradleTooling, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.compiler-plugin-support.gradle, descriptorPath=kotlin.compiler-plugins.compiler-plugin-support.gradle.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.gradleJava.compilerPlugin, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.gradle.gradle-java, descriptorPath=kotlin.gradle.gradle-java.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.gradleJava, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.gradle.code-insight-common, descriptorPath=kotlin.gradle.code-insight-common.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.gradleCodeInsightCommon, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.gradle.code-insight-toml, descriptorPath=kotlin.gradle.code-insight-toml.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.gradle.versionCatalog.toml, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.gradle.gradle, descriptorPath=kotlin.gradle.gradle.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.gradle, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.git, descriptorPath=kotlin.git.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.git, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.run-configurations.junit-fe10, descriptorPath=kotlin.run-configurations.junit-fe10.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.fe10.junit, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.jvm-debugger.sequence, descriptorPath=kotlin.jvm-debugger.sequence.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.debugger.sequence, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.coverage, descriptorPath=kotlin.coverage.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.coverage, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.copyright, descriptorPath=kotlin.copyright.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.copyright, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.scripting, descriptorPath=kotlin.compiler-plugins.scripting.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.scripting.gradleJava, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.sam-with-receiver.maven, descriptorPath=kotlin.compiler-plugins.sam-with-receiver.maven.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.samWithReceiver.maven, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.sam-with-receiver.gradle, descriptorPath=kotlin.compiler-plugins.sam-with-receiver.gradle.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.samWithReceiver.gradleJava, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.parcelize.gradle, descriptorPath=kotlin.compiler-plugins.parcelize.gradle.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.parcelize.gradleJava, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.noarg.maven, descriptorPath=kotlin.compiler-plugins.noarg.maven.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.noarg.maven, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.noarg.gradle, descriptorPath=kotlin.compiler-plugins.noarg.gradle.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.noarg.gradleJava, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.lombok.maven, descriptorPath=kotlin.compiler-plugins.lombok.maven.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.lombok.maven, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.lombok.gradle, descriptorPath=kotlin.compiler-plugins.lombok.gradle.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.lombok.gradleJava, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.kotlinx-serialization.maven, descriptorPath=kotlin.compiler-plugins.kotlinx-serialization.maven.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.kotlinxSerialization.maven, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.kotlinx-serialization.gradle, descriptorPath=kotlin.compiler-plugins.kotlinx-serialization.gradle.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.kotlinxSerialization.gradleJava, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.kapt, descriptorPath=kotlin.compiler-plugins.kapt.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.kapt.gradleJava, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.assignment.maven, descriptorPath=kotlin.compiler-plugins.assignment.maven.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.assignment.maven, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.assignment.common.k1, descriptorPath=kotlin.compiler-plugins.assignment.common.k1.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.assignment.k1, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.assignment.gradle, descriptorPath=kotlin.compiler-plugins.assignment.gradle.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.assignment.gradleJava, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.allopen.maven, descriptorPath=kotlin.compiler-plugins.allopen.maven.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.allopen.maven, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.compiler-plugins.allopen.gradle, descriptorPath=kotlin.compiler-plugins.allopen.gradle.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.compilerPlugin.allopen.gradleJava, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.code-insight.postfix-templates.k1, descriptorPath=kotlin.code-insight.postfix-templates.k1.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.codeInsight.postfix, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.code-insight.intentions.shared, descriptorPath=kotlin.code-insight.intentions.shared.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.codeInsight.intentions.shared, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.code-insight.inspections.shared, descriptorPath=kotlin.code-insight.inspections.shared.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.codeInsight.inspections.shared, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.base.injection, descriptorPath=kotlin.base.injection.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.base.injection, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.base.fe10.project-structure, descriptorPath=kotlin.base.fe10.project-structure.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.base.fe10.projectStructure, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.base.fe10.plugin, descriptorPath=kotlin.base.fe10.plugin.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.base.fe10.plugin, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.base.fe10.analysis-api-platform, descriptorPath=kotlin.base.fe10.analysis-api-platform.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.base.fe10.analysisApiPlatform, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.base.analysis-api-platform, descriptorPath=kotlin.base.analysis-api-platform.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin.idea.base.analysisApiPlatform, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin-ultimate.ultimate-native, descriptorPath=kotlin-ultimate.ultimate-native.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=com.jetbrains.mpp, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin-ultimate.common-native, descriptorPath=kotlin-ultimate.common-native.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=com.jetbrains.kotlin.commonNative, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, moduleName=kotlin.performanceExtendedPlugin, descriptorPath=kotlin.performanceExtendedPlugin.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=com.intellij.performance.performancePlugin, isBundled=true), PluginDescriptor(name=Java, id=com.intellij.java, moduleName=intellij.java.performancePlugin, descriptorPath=intellij.java.performancePlugin.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/java, version=242.20224.159, package=com.intellij.java.performancePlugin, isBundled=true), PluginDescriptor(name=Java, id=com.intellij.java, moduleName=intellij.java.compiler.charts, descriptorPath=intellij.java.compiler.charts.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/java, version=242.20224.159, package=com.intellij.java.compiler.charts, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.impl, descriptorPath=intellij.platform.vcs.impl.xml, path=/opt/jetbrains/idea-IU-242.20224.159/lib, version=242.20224.159, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.log.impl, descriptorPath=intellij.platform.vcs.log.impl.xml, path=/opt/jetbrains/idea-IU-242.20224.159/lib, version=242.20224.159, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.dvcs.impl, descriptorPath=intellij.platform.vcs.dvcs.impl.xml, path=/opt/jetbrains/idea-IU-242.20224.159/lib, version=242.20224.159, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.collaborationTools, descriptorPath=intellij.platform.collaborationTools.xml, path=/opt/jetbrains/idea-IU-242.20224.159/lib, version=242.20224.159, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.coverage, descriptorPath=intellij.platform.coverage.xml, path=/opt/jetbrains/idea-IU-242.20224.159/lib, version=242.20224.159, package=null, isBundled=true), PluginDescriptor(name=Java, id=com.intellij.java, moduleName=intellij.java.vcs, descriptorPath=intellij.java.vcs.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/java, version=242.20224.159, package=null, isBundled=true), PluginDescriptor(name=Java, id=com.intellij.java, moduleName=intellij.java.unscramble, descriptorPath=intellij.java.unscramble.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/java, version=242.20224.159, package=null, isBundled=true), PluginDescriptor(name=Java, id=com.intellij.java, moduleName=intellij.java.featuresTrainer, descriptorPath=intellij.java.featuresTrainer.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/java, version=242.20224.159, package=null, isBundled=true), PluginDescriptor(name=Java, id=com.intellij.java, moduleName=intellij.java.structuralSearch, descriptorPath=intellij.java.structuralSearch.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/java, version=242.20224.159, package=null, isBundled=true), PluginDescriptor(name=Terminal, id=org.jetbrains.plugins.terminal, descriptorPath=plugin.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/terminal, version=242.20224.159, package=org.jetbrains.plugins.terminal, isBundled=true), PluginDescriptor(name=Kotlin, id=org.jetbrains.kotlin, descriptorPath=plugin.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/Kotlin, version=242.20224.159-IJ, package=org.jetbrains.kotlin, isBundled=true), PluginDescriptor(name=Java, id=com.intellij.java, descriptorPath=plugin.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/java, version=242.20224.159, package=null, isBundled=true), PluginDescriptor(name=JUnit, id=JUnit, descriptorPath=plugin.xml, path=/opt/jetbrains/idea-IU-242.20224.159/plugins/junit, version=242.20224.159, package=null, isBundled=true), PluginDescriptor(name=Scala, id=org.intellij.scala, descriptorPath=plugin.xml, path=~/.config/JetBrains/RemoteDev-IU-242.20224.159/_home_adam_miskiewicz_repos_treehouse/plugins/Scala, version=2024.2.17, package=null, isBundled=false), )
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.kt:157)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	... 63 more
```

--- a/base/BUILD
+++ b/base/BUILD
@@ -11,12 +11,12 @@ load(
     "intellij_plugin_library",
     "stamped_plugin_xml",
 )
+load("//intellij_platform_sdk:build_defs.bzl", "java_version_flags", "select_for_plugin_api")
 load(
     "//testing:test_defs.bzl",
     "intellij_integration_test_suite",
     "intellij_unit_test_suite",
 )
-load("//intellij_platform_sdk:build_defs.bzl", "java_version_flags")
 
 java_library(
     name = "base",
@@ -50,7 +50,35 @@ java_library(
 
 intellij_plugin_library(
     name = "plugin_library",
-    plugin_xmls = ["src/META-INF/blaze-base.xml"],
+    plugin_xmls = ["src/META-INF/blaze-base.xml"] +  #api241
+                  select_for_plugin_api({
+                      "android-studio-2022.2": [],
+                      "android-studio-2022.3": [],
+                      "android-studio-2023.1": [],
+                      "android-studio-2023.2": [],
+                      "intellij-2022.1": [],
+                      "intellij-ue-2022.1": [],
+                      "intellij-2022.2": [],
+                      "intellij-ue-2022.2": [],
+                      "intellij-2022.3": [],
+                      "intellij-ue-2022.3": [],
+                      "intellij-2023.1": [],
+                      "intellij-ue-2023.1": [],
+                      "intellij-2023.2": [],
+                      "intellij-ue-2023.2": [],
+                      "intellij-2023.3": [],
+                      "intellij-ue-2023.3": [],
+                      "intellij-2024.1": [],
+                      "intellij-ue-2024.1": [],
+                      "clion-2022.1": [],
+                      "clion-2022.2": [],
+                      "clion-2022.3": [],
+                      "clion-2023.1": [],
+                      "clion-2023.2": [],
+                      "clion-2023.3": [],
+                      "clion-2024.1": [],
+                      "default": ["src/META-INF/dependency-on-lvcs-impl.xml"],
+                  }),
     visibility = PLUGIN_PACKAGES_VISIBILITY,
     deps = [":base"],
 )

--- a/base/src/META-INF/dependency-on-lvcs-impl.xml
+++ b/base/src/META-INF/dependency-on-lvcs-impl.xml
@@ -1,0 +1,20 @@
+<!--
+  ~ Copyright 2024 The Bazel Authors. All rights reserved.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~    http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<idea-plugin>
+    <dependencies>
+        <module name="intellij.platform.lvcs.impl"/>
+    </dependencies>
+</idea-plugin>

